user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserQuotas.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.user.operator;
+
+import io.vertx.core.json.JsonObject;
+import org.I0Itec.zkclient.ZkClient;
+import org.I0Itec.zkclient.serialize.BytesPushThroughSerializer;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.nio.charset.Charset;
+import java.util.Map;
+
+/**
+ * Utility class for managing Quotas for Kafka User
+ */
+public class KafkaUserQuotas {
+    private static final Logger log = LogManager.getLogger(KafkaUserQuotas.class.getName());
+
+    private final static int CONNECTION_TIMEOUT = 30_000;
+
+    private ZkClient zkClient;
+
+    public KafkaUserQuotas(String zookeeperUrl, int zookeeperSessionTimeout) {
+        zkClient = new ZkClient(zookeeperUrl, zookeeperSessionTimeout, CONNECTION_TIMEOUT, new BytesPushThroughSerializer());
+    }
+
+    /**
+     * Create or update the quotas for the given user.
+     *
+     * @param username The name of the user which should be created or updated
+     * @param quotas The desired user quotas
+     */
+    public void createOrUpdate(String username, JsonObject quotas) {
+        byte[] data = zkClient.readData("/config/users/" + username, true);
+
+        if (data != null)   {
+            log.debug("Updating quotas for user {}", username);
+            zkClient.writeData("/config/users/" + username, updateUserJson(data, quotas));
+        } else {
+            log.debug("Creating quotas for user {}", username);
+            ensurePath("/config/users");
+            zkClient.createPersistent("/config/users/" + username, createUserJson(quotas));
+        }
+
+        notifyChanges(username);
+    }
+
+    /**
+     * Generates the JSON with the credentials
+     *
+     * @param quotas  quotas
+     * @return  Returns the generated JSON as byte array
+     */
+    protected byte[] createUserJson(JsonObject quotas)   {
+        JsonObject json = new JsonObject()
+                .put("version", 1)
+                .put("config", new JsonObject());
+
+        for (Map.Entry<String, Object> quota: quotas.getMap().entrySet()) {
+            json.getJsonObject("config").put(quota.getKey(), quota.getValue().toString());
+        }
+
+        return json.encode().getBytes(Charset.defaultCharset());
+    }
+
+    /**
+     * Updates the quotas in existing JSON
+     *
+     * @param user JSON string with existing user configuration as byte[]
+     * @param quotas  quotas
+     *
+     * @return  Returns the updated JSON as byte array
+     */
+    protected byte[] updateUserJson(byte[] user, JsonObject quotas)   {
+        JsonObject json = new JsonObject(new String(user, Charset.defaultCharset()));
```

Comment:
Using the default charset is wrong. What does Kafka use to encode the String? Likely its either ASCII or UTF-8.